core/src/autogluon/core/scheduler/seq_scheduler.py

```diff
@@ -0,0 +1,273 @@
+
+        time_start = time.time()
+
+        avg_trial_run_time = None
+
+        r = range(self.num_trials)
+        for i in (tqdm(r) if self.num_trials < 1000 else r):
+            trial_start_time = time.time()
+            self.run_trial(task_id=i)
+            trial_end_time = time.time()
+
+            if self.max_reward and self.get_best_reward() >= self.max_reward:
+                logger.log(20, f'\tMax reward is reached')
+                break
+
+            if self.time_out is not None:
+                avg_trial_run_time = self.get_average_trial_time_(i, avg_trial_run_time, trial_start_time, trial_end_time)
+                if not self.has_enough_time_for_trial_(self.time_out, time_start, trial_start_time, trial_end_time, avg_trial_run_time):
+                    logger.log(20, f'\tTime limit exceeded...')
+                    break
+
+    @classmethod
+    def has_enough_time_for_trial_(cls, time_out, time_start, trial_start_time, trial_end_time, avg_trial_run_time, fill_factor=0.95):
+        """
+        Checks if the remaining time is enough to run another trial.
+
+        Parameters
+        ----------
+        time_out total
+            timeout in m
+        time_start
+            trials start time
+        trial_start_time
+            last trial start time
+        trial_end_time
+            last trial end time
+        avg_trial_run_time
+            running average of all trial runs
+        fill_factor: float
+            discount of `avg_trial_run_time` allowed for a next trial. Default is 0.95 of `avg_trial_run_time`
+
+        Returns
+        -------
+            True if there is enough time to run another trial give runs statistics and remaining time
+
+        """
+        time_spent = trial_end_time - time_start
+        is_timeout_exceeded = time_spent >= time_out
+        time_left = time_start + time_out - trial_end_time
+        is_enough_time_for_another_trial = True
+        if avg_trial_run_time:
+            is_enough_time_for_another_trial = time_left > avg_trial_run_time * fill_factor
+        return is_enough_time_for_another_trial and not is_timeout_exceeded
+
+    @classmethod
+    def get_average_trial_time_(cls, i, avg_trial_run_time, trial_start_time, time_end):
+        trial_time = time_end - trial_start_time
+        if avg_trial_run_time is None:
+            avg_trial_run_time = trial_time
+        else:
+            avg_trial_run_time = ((avg_trial_run_time * i) + trial_time) / (i + 1)
+        return avg_trial_run_time
+
+    def run_trial(self, task_id=0):
+        searcher_config = self.searcher.get_config()
+        reporter = LocalReporter(task_id, searcher_config, self.training_history, self.config_history)
+        task_config = deepcopy(EasyDict(self.train_fn.kwvars))
+        task_config['task_id'] = task_id
+        self.searcher.register_pending(searcher_config)
+        self.train_fn(task_config, config=searcher_config, reporter=reporter)
```

Comment:
The signature of train_fn is train_fn(args, reporter), where args is a dict which is the union of config (as dict) and other things (optional). So this will not work here.

Reply:
I think it needs something like
   task_config.update(searcher_config)

Reply:
It is wrapped in `@args` decorator, so the wrapper takes care into translating searcher config into actual valies